domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
+			modelUUIDs[i] = modelUUID
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There should be no errors returned if there are no activated models found.
+		if err != nil {
+			return nil, err
+		}
+
+		activatedModelUUIDToChangeEventMap := make(map[string]struct{})
+		for _, activatedModelUUID := range activatedModelUUIDs {
+			activatedModelUUIDToChangeEventMap[activatedModelUUID.String()] = struct{}{}
+		}
+
+		activatedModelChangeEvents := make([]changestream.ChangeEvent, 0, len(changes))
```

Comment:
This should be `len(activatedModelUUIDS)`

Reply:
CMIIW, but if we have a name update followed by a true status update for the same model uuid, and if we set the change event cap to len(activatedModelUUIDS), a new slice for be allocated for this right? Happy to change it if you feel that the frequency is low and the pros of saving mem outweights the con of allocating a new slice :)

Reply:
Good point. Which ever way you look at it you run the risk of over or under allocating.